packages/typespec-client-generator-core/src/example.ts

```diff
@@ -0,0 +1,630 @@
+        }
+      } else if (name === "headers") {
+        for (const subName of Object.keys(example.headers)) {
+          const header = response.headers.find((p) => p.serializedName === subName);
+          if (header) {
+            const value = diagnostics.pipe(
+              getSdkTypeExample(header.type, example[name][subName], relativePath)
+            );
+            if (value) {
+              responseExample.headers.push({
+                header,
+                value,
+              });
+            }
+          } else {
+            addExampleValueNoMappingDignostic(
+              diagnostics,
+              { [subName]: example[name][subName] },
+              relativePath
+            );
+          }
+        }
+      } else {
+        addExampleValueNoMappingDignostic(diagnostics, { [name]: example[name] }, relativePath);
+      }
+    }
+  }
+  return diagnostics.wrap(responseExample);
+}
+
+function getSdkTypeExample(
+  type: SdkType | SdkModelPropertyType,
+  example: any,
+  relativePath: string
+): [SdkTypeExample | undefined, readonly Diagnostic[]] {
+  const diagnostics = createDiagnosticCollector();
+
+  if (isSdkIntKind(type.kind) || isSdkFloatKind(type.kind)) {
+    return getSdkBaseTypeExample("number", type as SdkType, example, relativePath);
+  } else if (type.kind === "string" || type.kind === "bytes") {
+    return getSdkBaseTypeExample("string", type as SdkType, example, relativePath);
+  } else if (type.kind === "boolean") {
+    return getSdkBaseTypeExample("boolean", type as SdkType, example, relativePath);
+  } else if (
+    type.kind === "password" ||
+    type.kind === "guid" ||
+    type.kind === "url" ||
+    type.kind === "uri" ||
+    type.kind === "ipAddress" ||
+    type.kind === "uuid" ||
+    type.kind === "ipV4Address" ||
+    type.kind === "ipV6Address" ||
+    type.kind === "eTag" ||
+    type.kind === "armId" ||
+    type.kind === "azureLocation" ||
+    type.kind === "plainDate" ||
+    type.kind === "plainTime"
+  ) {
+    return getSdkBaseTypeExample("string", type as SdkType, example, relativePath);
+  } else if (type.kind === "nullable") {
+    if (example === null) {
+      return diagnostics.wrap({
+        kind: "null",
+        type,
+        value: null,
+      } as SdkNullExample);
+    } else {
+      return getSdkTypeExample(type.type, example, relativePath);
+    }
+  } else if (type.kind === "any") {
+    return diagnostics.wrap({
+      kind: "any",
+      type,
+      value: example,
+    } as SdkAnyExample);
+  } else if (type.kind === "constant") {
+    if (example === type.value) {
+      return getSdkBaseTypeExample(
+        typeof type.value as "string" | "number" | "boolean",
+        type,
+        example,
+        relativePath
+      );
+    } else {
+      addExampleValueNoMappingDignostic(diagnostics, example, relativePath);
+      return diagnostics.wrap(undefined);
+    }
+  } else if (type.kind === "enum") {
```

Comment:
i think using a `switch` statement instead of a bunch of `else if`s would be cleaner